src/viking.js

```diff
@@ -1,11 +1,129 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+  this.health = health;
+  this.strength = strength;
+}
+Soldier.prototype.attack = function () {
+    return this.strength;
+};
+
+Soldier.prototype.receiveDamage = function (damage) {
+    this.health = this.health - damage;
+}
+
+
 
 // Viking
-function Viking() {}
+function Viking(name, health, strength) {
+    Soldier.call(this, health, strength);
+    this.name = name;
+}
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.constructor = Viking;
+
+Viking.prototype.receiveDamage = function (damage) {
+this.health = this.health - damage;
+if (this.health > 0 ) {
+    return `${this.name} has received ${damage} points of damage`;
+} else {
+    return `${this.name} has died in act of combat`;
+}
+};
+
+Viking.prototype.battleCry = function () {
+    return "Odin Owns You All!"
+}
+
 
 // Saxon
-function Saxon() {}
+function Saxon(health, strength) {
+    Soldier.call(this, health, strength);
+}
+Saxon.prototype = Object.create(Soldier.prototype);
+Saxon.prototype.constructor = Saxon;
+
+
+Saxon.prototype.receiveDamage = function(theDamage) {
+    this.health = this.health - theDamage;
+    if(this.health > 0) {
+        return `A Saxon has received ${theDamage} points of damage`;
+    } else {
+        return "A Saxon has died in combat";
+    }
+}
+
 
 // War
-function War() {}
+function War() {
+this.vikingArmy = [];
+this.saxonArmy = [];
+}
+War.prototype.addViking = function (v) {
+    this.vikingArmy.push(v);
```

Comment:
Try to name your variable by something more significative, viking instead of v and saxon instead of s, even if it's longer, it's a good habit to do it ;)